drivers/shared/src/vic_log.c

```diff
@@ -29,80 +29,81 @@
  *****************************************************************************/
 
 #include <time.h>
-
 #include <vic_def.h>
 #include <vic_run.h>
 #include <vic_driver_shared.h>
 
 /******************************************************************************
- * @brief    Get string of current date and time.  Format YYMMDD-SSSSS.
+ * @brief    Finalize logging
  *****************************************************************************/
-char*
-get_current_datetime()
+void
+finalize_logging(void)
 {
-    const int size = 20;
-    char     *cdt = (char*)malloc(sizeof(char) * size);
-    char      ymd[8];
-    unsigned  seconds_since_midnight;
+    extern FILE *LOG_DEST;
 
-    if (cdt == NULL) {
-        return NULL;
+    if (!(LOG_DEST == stdout || LOG_DEST == stderr)) {
+        fclose(LOG_DEST);
     }
+}
 
-    memset(cdt, 0, size);
-
-    time_t currDateTime;
-    currDateTime = time(NULL);
+/******************************************************************************
+ * @brief    Get string of current date and time.  Format YYMMDD-SSSSS.
+ *****************************************************************************/
+void
+get_current_datetime(char *cdt)
```

Comment:
To make memory management easier (memory allocated in the calling function), pass a pointer as a function argument rather than return a pointer.